test/jdk/java/beans/Introspector/DefaultMethodBeanPropertyTest.java

```diff
@@ -144,60 +154,288 @@ public NavigableSet<?> getFoo() {
+            "public void DefaultMethodBeanPropertyTest$A9.setFoo2(java.lang.Integer)"
         );
     }
 
 // Helper methods
 
-    public static void verifyProperties(Class<?> type, String... getterNames) {
-
-        // Gather expected properties
-        final HashSet<PropertyDescriptor> expected = new HashSet<>();
-        for (String methodName : getterNames) {
-            final String suffix = methodName.substring(3);
-            final String propName = Introspector.decapitalize(suffix);
-            final Method getter;
-            try {
-                getter = type.getMethod(methodName);
-            } catch (NoSuchMethodException e) {
-                throw new Error("unexpected error", e);
-            }
-            final PropertyDescriptor propDesc;
-            try {
-                propDesc = new PropertyDescriptor(propName, getter, null);
-            } catch (IntrospectionException e) {
-                throw new Error("unexpected error", e);
-            }
-            expected.add(propDesc);
-        }
-
-        // Verify properties can be found directly
-        expected.stream()
-                .map(PropertyDescriptor::getName)
-                .filter(name -> BeanUtils.getPropertyDescriptor(type, name) == null)
-                .findFirst()
-                .ifPresent(name -> {
-                    throw new Error("property \"" + name + "\" not found in " + type);
-                });
-
-        // Gather actual properties
-        final Set<PropertyDescriptor> actual =
-                Set.of(BeanUtils.getPropertyDescriptors(type));
-
-        // Verify the two sets are the same
+    private static void verifyEquality(String title, Set<String> expected, Set<String> actual) {
         if (!actual.equals(expected)) {
-            throw new Error("mismatch: " + type
-              + "\nACTUAL:\n  "
-              + actual.stream()
-                      .map(Object::toString)
-                      .collect(Collectors.joining("\n  "))
-              + "\nEXPECTED:\n  "
-              + expected.stream()
-                        .map(Object::toString)
-                        .collect(Collectors.joining("\n  ")));
+            throw new Error(title + " mismatch: "
+                    + "\nACTUAL:\n  "
+                    + actual.stream()
+                            .map(Object::toString)
+                            .collect(Collectors.joining("\n  "))
+                    + "\nEXPECTED:\n  "
+                    + expected.stream()
+                            .map(Object::toString)
+                            .collect(Collectors.joining("\n  ")));
+        }
+    }
+
+    public static void verifyProperties(Class<?> type,  String... methodNames) {
+        try {
+            final Set<String> expected = new HashSet<>(Arrays.asList(methodNames));
+            final Set<String> actual = Arrays
+                    .stream(Introspector.getBeanInfo(type).getPropertyDescriptors())
```

Comment:
```suggestion
                    .stream(Introspector.getBeanInfo(type)
                                        .getPropertyDescriptors())
```
It may be better this way, the call to `getPropertyDescriptors` is clearly visible.